test/fixtures/sidecars/volume-mounts.yaml

```diff
@@ -0,0 +1,35 @@
+---
+name: volume-mounts
+env:
+  - name: DATACENTER
+    value: foo
+  - name: FROM_INJECTOR
+    value: bar
+containers:
+- name: sidecar-nginx
+  image: nginx:1.12.2
+  imagePullPolicy: IfNotPresent
+  env:
+    - name: DATACENTER
+      value: bf2
+  ports:
+    - containerPort: 80
+  volumeMounts:
+    - name: nginx-conf
+      mountPath: /etc/nginx
+    - name: test-vol
+      mountPath: /tmp/test
+- name: another-sidecar
+  image: foo:69
+  ports:
+    - containerPort: 420
+  volumeMounts:
```

Comment:
you should test the cases where:
* a sidecar container already has a volumeMount, and we add another one.
* a sidecar container has no VolumeMounts, and we add the first one
* a sidecar container has a volumeMount named "test1", and we attempt to add "test1", and skip adding it

Reply:
Updated in the latest commit, but how is it actually tested that it will end up in the deployed YAML as intended?

Reply:
Ah, we do not have a way (yet!) to call our HTTP handler with a fixture payload of a pod, and test the response body. If you have bandwidth and would like to investigate how to set this up, that would be amazing!

If not, this works :)

Reply:
Now I understand it :D 
I'll see if I can come up with something in another PR :+1:

Reply:
awesome, that would be a huge boon!